index.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Create a middleware handler for collecting statistics.
+ *
+ * @callback {function} recordBuilder The builder function is called
+ *   for each request, the return value is collected as a stats record.
+ * @param {Request} req The express request object.
+ * @param {Response} res The express response object.
+ * @return {Object} A key/value map describing a stats record.
+ * @end
+ *
+ * @header xstats(recordBuilder)
+ */
+module.exports = createStatsHandler;
+
+/**
+ * Register an observer function to be notified whenever a new stats record
+ * is collected.
+ * @callback {function} observer The observer function.
+ * @param {Object} data The data returned by the record builder function.
+ * @end
+ *
+ * @header xstats.onRecord(observer)
+ */
+module.exports.onRecord = observe;
```

Comment:
> I'd call function observe(), function onRecord(), I think its easier to follow (observe sounds like the function passed in as an argument to onRecord).

Fixed.

> But given that what you have implemented in observe() is "event emitter lite", why not just make the xstat module an event emitter?

1. Micro-optimization. Event emitter uses `fn.call(this, arg1)`, which is about 50% slower than `fn(arg1)`. And we don't need `this` object.

2. The extra API like `removeListener` and support for multiple events is not needed now.

> Which allows you, in the future, to implement optimizations such as not bothering to emit a record when there are no listeners. And maybe to not even bother calling the record builder function if there are no listeners.

That can be already implemented now via `if (!observers)`.

---

I don't care that much about the implementation. If you still think EventEmitter is a better solution, then I'll rework this patch to get it landed and move on.